tools/cli/commands/clean.js

```diff
@@ -0,0 +1,607 @@
+ * External dependencies
+ */
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import child_process from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject, { promptForType } from '../helpers/promptForProject';
+import { normalizeCleanArgv } from '../helpers/normalizeArgv';
+import { runCommand } from '../helpers/runCommand';
+import { allProjects } from '../helpers/projectHelpers';
+import fs from 'fs';
+
+/**
+ * Command definition for the build subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ *
+ * @returns {object} Yargs with the build commands defined.
+ */
+export function cleanDefine( yargs ) {
+	yargs.command(
+		'clean [project]',
+		'Removes unversioned files and folder from a specific project.',
+		yarg => {
+			yarg
+				.positional( 'project', {
+					describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+					type: 'string',
+				} )
+				.positional( 'include', {
+					alias: 'i',
+					describe: 'Files/folders to include for deletion',
+					type: 'string',
+					choices: [ 'node_modules', 'composer.lock', 'vendor', 'working', 'ignored' ],
+				} )
+				.option( 'all', {
+					alias: 'a',
+					type: 'boolean',
+					description: 'Remove everything from monorepo root',
+				} )
+				.option( 'dist', {
+					type: 'boolean',
+					description: 'Remove distributed files (vendor, node_modules)',
+				} );
+		},
+		async argv => {
+			await cleanCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handle args for clean command.
+ *
+ * @param {argv}  argv - the arguments passed.
+ */
+export async function cleanCli( argv ) {
+	if ( argv.project ) {
+		await parseProj( argv );
+	} else {
+		await promptProj( argv );
+	}
+
+	if ( argv.include ) {
+		await parseToClean( argv );
+	} else {
+		await promptForClean( argv );
+	}
+
+	const allFiles = await collectAllFiles( argv.toClean );
+	const toCleanFiles = await collectCleanFiles( allFiles, argv.toClean );
+	const runConfirm = await confirmRemove( argv );
+
+	if ( ! runConfirm.confirm ) {
+		console.log( chalk.red( 'Cancelling jetpack clean.' ) );
+		return;
+	}
+
+	await cleanFiles( toCleanFiles, argv );
+	return;
+}
+
+/**
+ * Delete the files that we want.
+ *
+ * @param {Array} toCleanFiles - files that we want to clean.
+ * @param {object} argv - the arguments passed.
+ */
+async function cleanFiles( toCleanFiles, argv ) {
+	for ( const file of toCleanFiles ) {
+		fs.rm( file, { recursive: true, force: true }, ( err ) => {
+			console.log( 'Cleaning ', file );
+			if ( err ) {
+				// File deletion failed
+				console.error( err.message );
+				return;
+			}
+		} );
+	}
+
+	// Cleanup root and tools node_modules folder if that's what we're deleting.
+	if ( argv.scope === 'all' && argv.toClean.includes( 'node_modules' ) ) {
+		process.on( 'exit', async () => {
+			await runCommand( 'rm', [ '-rf', 'node_modules', 'tools/cli/node_modules' ] );
+		} );
+	}
```

Comment:
From what I can see, `fs.rmSync` removes all other node_modules directories properly, it's just `/tools/cli/node_modules` and `/node_modules` that don't get removed (which would only be queued for removal when the scope is the entire monorepo with `all`), unless I'm missing some other cases. 

> Also of note is that the 'exit' handler is not allowed to be async.

Good to know!

Reply:
`fs.rmSync` removes all the node_modules directories properly. The problem is that `esm` recreates them it its own 'exit' handler (see [here](https://github.com/standard-things/esm/blob/511d672ae13e8bee13ba19bd7fdc2a3206c9d7d7/src/caching-compiler.js#L168-L404)).

My code suggestion just goes ahead and re-deletes all of the selected node_modules dirs instead of trying to guess which ones might be recreated.

Reply:
> My code suggestion just goes ahead and re-deletes all of the selected node_modules dirs instead of trying to guess which ones might be recreated.

Perfect, works for me. Should be in 38ce25b